integration_tests/run_pyspark_from_build.sh

```diff
@@ -310,6 +312,9 @@ EOF
         # total memory usage, especially host memory usage because when copying data to the GPU
         # buffers as large as batchSizeBytes can be allocated, and the fewer of them we have the better.
         LOCAL_PARALLEL=$(( $CPU_CORES > 4 ? 4 : $CPU_CORES ))
+        # The only case where we want worker logs is in local mode so we set the value here explicitly
+        # We can't use the PYSP_TEST_spark_master as it's not always set e.g. when using --master
+        export USE_WORKER_LOGS=1
```

Comment:
why does it not apply to L306 dealing with the multi-executor spark that  can also be executed in non-xdist mode?

Reply:
I wanted to restrict the focus of this PR to nightly tests run only which run in local mode. Do you feel strongly about adding it for this flow as well?

Reply:
Yes, the logic should be as straightforward as possible. The only distinction I see required is whether we have a Spark app running by an xdist worker or not.

Reply:
I have added logs to local-cluster mode as well. PTAL